server/services/receive.js

```diff
@@ -0,0 +1,166 @@
+const bunyan = require('bunyan');
+
+const log = bunyan.createLogger({ name: 'receive-service' });
+
+const path = require('path');
+
+const Response = require(path.join(__dirname, './response'));
+
+const GraphAPi = require(path.join(__dirname, './graph-api'));
+
+const Rescue = require(path.join(__dirname, './rescue'));
+
+const Update = require(path.join(__dirname, './update'));
+
+const has = require('has');
+
+module.exports = class Receive {
+  constructor(user, webhookEvent) {
+    this.user = user;
+    this.webhookEvent = webhookEvent;
+  }
+
+  handleMessage() {
+    const event = this.webhookEvent;
+
+    let responses = [];
+
+    try {
+      if (event.message) {
+        const { message } = event;
+
+        if (message.text) {
+          responses = this.handleTextMessage();
+        } else if (message.attachments) {
+          responses = this.handleAttachmentMessages();
+        } else if (message.quick_reply) {
+          responses = this.handleQuickReply();
+        }
+      } else if (event.postback) {
+        responses = this.handlePostback();
+      }
+    } catch (error) {
+      log.error(error);
+      responses = {
+        text: `An error has occured: '${error}'. We have been notified and \
+        will fix the issue shortly!`,
+      };
+    }
+
+    if (Array.isArray(responses)) {
+      let delay = 0;
+      responses.forEach((response) => {
+        this.sendMessage(response, delay * 2000);
+        delay += 1;
+      });
+    } else {
+      this.sendMessage(responses);
+    }
+  }
+
+  handleTextMessage() {
+    log.info('Received text:', `${this.webhookEvent.message.text} for ${this.user}`);
+    const message = this.webhookEvent.message.text.toLowerCase();
+
+    let response = Response.genStandardErrorMessage();
+
+    if (message.includes('start over')) {
+      response = Response.genStartMessage();
+    } else if (message.includes('#')) {
+      const update = new Update(this.user, this.webhookEvent);
+      response = update.handleUpdate();
+    } else if (this.user.step) {
+      const rescue = new Rescue(this.user, this.webhookEvent);
+      response = rescue.handleStep();
+    }
+
+    return response;
+  }
+
+  handleAttachmentMessages() {
+    let response = Response.genStandardErrorMessage();
+
+    // Get the attachment
+    const attachment = this.webhookEvent.message.attachments[0];
+    log.info('Received attachment:', attachment);
+
+    const rescue = new Rescue(this.user, this.webhookEvent);
+    response = rescue.handleAttachment(attachment);
+
+    return response;
+  }
+
+  handleQuickReply() {
+    const { payload } = this.webhookEvent.message.quick_reply;
+
+    return this.handlePayload(payload);
+  }
+
+  handlePostback() {
+    const { postback } = this.webhookEvent;
+    const { payload } = postback;
```

Comment:
Combine into one destructuring since postback is not being used

`const { payload } = this.webhookEvent.postback`